models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -798,24 +798,30 @@ public function preGetData(mixed $container, array $params = []): mixed
         $params['fieldname'] = $this->getName();
         if ($container instanceof DataObject\Concrete) {
             $data = $container->getObjectVar($this->getName());
-            if ($this->getLazyLoading() && !$container->isLazyKeyLoaded($this->getName())) {
-                $data = $this->load($container, $params);
-
-                $setter = 'set' . ucfirst($this->getName());
-                if (method_exists($container, $setter)) {
-                    $container->$setter($data);
-                    $this->markLazyloadedFieldAsLoaded($container);
-                }
-            }
-            $this->preSetData($container, $data, $params);
         } elseif ($container instanceof DataObject\Localizedfield) {
             $data = $params['data'];
         } elseif ($container instanceof DataObject\Fieldcollection\Data\AbstractData) {
             $data = $container->getObjectVar($this->getName());
         } elseif ($container instanceof DataObject\Objectbrick\Data\AbstractData) {
+            $params['context'] = [
+                'object' => $params['owner']->getObject(),
+                'brickField' => $params['fieldname'],
+                'containerKey' => $params['owner']->getType(),
+                'fieldname' => $params['owner']->getFieldname(),
+            ];
             $data = $container->getObjectVar($this->getName());
         }
 
+        if ($this->getLazyLoading() && !$container->isLazyKeyLoaded($this->getName())) {
+            $data = $this->load($container, $params);
+
+            $setter = 'set' . ucfirst($this->getName());
+            if (method_exists($container, $setter)) {
+                $container->$setter($data);
+            }
+        }
+        $this->preSetData($container, $data, $params);
```

Comment:
that now gets executed for every container type, not just for concretes. Is that  a problem?

Reply:
`$this->markLazyloadedFieldAsLoaded($container);` is missing too

Reply:
The `markLazyloadedFieldAsLoaded` is done within `preSetData`

https://github.com/pimcore/pimcore/blob/ee814a43916bd80cd44082e0cef6902257be5499/models/DataObject/ClassDefinition/Data/Block.php#L705-L707

and it felt like it's scanning through `localizedfields` which won't hurt for the other container types?
https://github.com/pimcore/pimcore/blob/ee814a43916bd80cd44082e0cef6902257be5499/models/DataObject/ClassDefinition/Data/Block.php#L709-L710